src/WebJobs.Extensions.DurableTask/Correlation/TelemetryActivator.cs

```diff
@@ -93,7 +96,18 @@ private void SetUpTelemetryClient()
             telemetryInitializer.ExcludeComponentCorrelationHttpHeadersOnDomains.Add("127.0.0.1");
             config.TelemetryInitializers.Add(telemetryInitializer);
 
-            config.InstrumentationKey = Environment.GetEnvironmentVariable("APPINSIGHTS_INSTRUMENTATIONKEY");
+            if (Environment.GetEnvironmentVariable("APPINSIGHTS_INSTRUMENTATIONKEY") != null)
```

Comment:
If you inject `INameResolver` into the constructor of this service, you can use that instead of `Environment.GetEnvironmentVariable()`.

That makes mocking environment variables for the tests possible.